Eatery/Controllers/Eateries/EateriesViewController.swift

```diff
@@ -468,62 +470,72 @@ extension EateriesViewController: UICollectionViewDataSource {
-                        attributedString.addAttributes([NSAttributedStringKey.foregroundColor : UIColor.darkGray, NSAttributedStringKey.font : UIFont.boldSystemFont(ofSize: 11.0)], range: match.range)
-                    }
-                } catch {
-                    NSLog("Error in handling regex")
+        if searchBar.text != "", let names = searchedMenuItemNames[eatery] {
+            let baseString = names.joined(separator: "\n")
+            let attributedString = NSMutableAttributedString(string: baseString,
+                                                             attributes: [.foregroundColor : UIColor.gray,
+                                                                          .font : UIFont.systemFont(ofSize: 11)])
+
+            do {
+                let regex = try NSRegularExpression(pattern: searchBar.text ?? "", options: .caseInsensitive)
+                for match in regex.matches(in: baseString,
+                                           options: [],
+                                           range: NSRange(location: 0, length: baseString.utf16.count)) {
+                                            attributedString.addAttributes([.foregroundColor : UIColor.darkGray,
+                                                                            .font : UIFont.boldSystemFont(ofSize: 11)],
+                                                                           range: match.range)
                 }
-                cell.menuTextView.attributedText = attributedString
-                cell.menuTextViewHeight.constant = cell.frame.height - 54.0
-            } else {
-                cell.menuTextView.text = nil
-                cell.menuTextViewHeight.constant = 0.0
+            } catch {
+                NSLog("Error in handling regex")
             }
+
+            cell.menuTextView.attributedText = attributedString
+            cell.isMenuTextViewVisible = true
         } else {
-            cell.menuTextViewHeight.constant = 0.0
+            cell.isMenuTextViewVisible = false
         }
 
         return cell
     }
 
     func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
-        let view = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: "HeaderView", for: indexPath) as! EateriesCollectionViewHeaderView
+        let header = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: "HeaderView", for: indexPath) as! EateriesCollectionViewHeaderView
 
         let (section, eateries) = data(for: indexPath.section)
 
-        if section == "Favorites" {
-            view.titleLabel.text = "Favorites"
-            view.titleLabel.textColor = .eateryBlue
-        } else if section == "Open" {
+        switch section {
+        case "Favorites":
```

Comment:
does not have to be today, but maybe during the refactor period we should have the sections as enums instead of strings